pkg/cmd/server/api/types.go

```diff
@@ -144,6 +144,24 @@ type ProjectConfig struct {
 	// It is in the format namespace/template and it is optional.
 	// If it is not specified, a default template is used.
 	ProjectRequestTemplate string
+
+	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
+	SecurityAllocator *SecurityAllocator
+}
+
+type SecurityAllocator struct {
+	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
+	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
+	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
+	// ranges Docker images will use once user namespaces are started).
+	UIDAllocatorRange string
+	// MCSAllocatorRange defines the range of MCS categories that will be assigned to namespaces. The format is
+	// "<prefix>/<numberOfLabels>[,<maxCategory>]". The default is "s0/2" and will allocate from c0 -> c1023, which means a total of 535k labels
+	// are available. If this value is changed after startup, projects will have different.
+	MCSAllocatorRange string
+	// MCSLabelsPerProject defines the number of labels that should be reserved per project. The default is 5 to match the default UID and MCS
```

Comment:
It's just to divide the convenient range sizes to match.  100k namespaces should be enough for anyone.

> On Jun 8, 2015, at 8:09 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/server/api/types.go:
> 
> > +
> > +	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
> > +	SecurityAllocator *SecurityAllocator
> > +}
> > +
> > +type SecurityAllocator struct {
> > +	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
> > +	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
> > +	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
> > +	// ranges Docker images will use once user namespaces are started).
> > +	UIDAllocatorRange string
> > +	// MCSAllocatorRange defines the range of MCS categories that will be assigned to namespaces. The format is
> > +	// "<prefix>/<numberOfLabels>[,<maxCategory>]". The default is "s0/2" and will allocate from c0 -> c1023, which means a total of 535k labels
> > +	// are available. If this value is changed after startup, projects will have different.
> > +	MCSAllocatorRange string
> > +	// MCSLabelsPerProject defines the number of labels that should be reserved per project. The default is 5 to match the default UID and MCS
> Does this value need to match or is matching just a cool default?
> 
> —
> Reply to this email directly or view it on GitHub.
>